### ISSUE TITLE
Match asking for account credentials even though they are already provided

### ISSUE THREAD

**Issue by Katsz** [fastlane/fastlane#17153]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

Duplicates #17152 in more details

### Match asking for account credentials even though they are already provided
After updating fastlane from 2.147.0 to 2.157.3 match start to ask Apple Developer Program account credentials even though they are already provided via `username:`. Issue was introduced in 2.157.3 as it works fine after rolling back to 2.157.2.

##### Command executed
```ruby
# Where matchConfig is `Struct`
match(username: matchConfig.username,
          git_url: GIT_URL,
          git_branch: matchConfig.developerTeamID,
          team_id: matchConfig.developerTeamID,
          app_identifier: matchConfig.bundleIdentifier,
          force_for_new_devices: matchConfig.signType == "development",
          type: matchConfig.signType,
          readonly: readonly,
	  git_user_email: matchConfig.username,
          git_full_name: matchConfig.fullName
    )
```

##### Complete output when running fastlane, including the stack trace and command used
![image](https://user-images.githubusercontent.com/16171550/92213478-48170b00-ee9c-11ea-99a0-f138507072ae.png)
![image](https://user-images.githubusercontent.com/16171550/92213502-4f3e1900-ee9c-11ea-8dab-22c78af4285a.png)

This problem was introduced here:
![image](https://user-images.githubusercontent.com/16171550/92216617-35520580-ee9f-11ea-8096-ddae2507f2bf.png)
[https://github.com/fastlane/fastlane/compare/2.157.2...2.157.3](https://github.com/fastlane/fastlane/compare/2.157.2...2.157.3#diff-137915dacc1c0b12e6d57863788d2c59L20-L24)
(I'm unaware how to embed diff fragment inside issue so bear with me)

Notice how `.login(user)` was used in 2.157.2 and not in 2.157.3

**Comment by Katsz** [comment]
Hey @joshdholtz 👋 , don't mind taking a look?

**Comment by stefanomondino** [comment]
same problem here. 2.157.2 is working but 2.157.3 isn't.

**Comment by rmtchop** [comment]
+1

**Comment by joshdholtz** [comment]
Looking into! Sorry about issue 😔

**Comment by joshdholtz** [comment]
I think I got a fix but I'm doing a bit more testing. I should have a PR out in about 20 minutes for testing. Sorry about about the issue 💔

**Comment by joshdholtz** [comment]
Hey fam! Got a fix in PR #17160. Instructions below if you are able to test if this fixes it for you 👇 

My apologies and ❤️s for causing this!

### Testing Steps 

Update `Gemfile` and run `bundle update fastlane`, `bundle update`, or `bundle install`

```rb
gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "joshdholtz-match-fix-team-select-and-improve-spaceship-connect-team-select"
```

**Comment by ChristopherDrum** [comment]
Hmm... I'm having this exact same issue, which started with my submission on September 25.
This past Friday the issue occurred with Fastlane v2.162.0
Not entirely sure which version started the problem for me.
I will open a new issue, as requested above.
